hscontrol/types/config.go

```diff
@@ -287,6 +288,7 @@ func LoadConfig(path string, isFile bool) error {
 
 	viper.SetDefault("dns.magic_dns", true)
 	viper.SetDefault("dns.base_domain", "")
+	viper.SetDefault("dns.override_local_dns", true)
```

Comment:
I think https://github.com/juanfont/headscale/pull/2438/files#r2046953908 is correct. I also just ran into this and needed to turn local dns override off.